pytorch_lightning/callbacks/progress.py

```diff
@@ -293,7 +294,9 @@ def init_test_tqdm(self) -> tqdm:
     def on_sanity_check_start(self, trainer, pl_module):
         super().on_sanity_check_start(trainer, pl_module)
         self.val_progress_bar = self.init_sanity_tqdm()
-        self.val_progress_bar.total = convert_inf(trainer.num_sanity_val_steps * len(trainer.val_dataloaders))
+        self.val_progress_bar.total = sum(
+            min(trainer.num_sanity_val_steps, len(d) if has_len(d) else float('inf')) for d in trainer.val_dataloaders
+        )
```

Comment:
@awaelchli `num_sanity_val_steps` should be independent of `limit_val_batches`(float)?

Reply:
if `num_sanity_val_steps=2`, `len(val_dataloader)=10` and `limit_val_batches=0.1`, should it run for 2 val_steps or 1?

Reply:
is this relevant here? I thought this pr is just about displaying the num_sanity steps that the trainer returns. 
if limit_val_batches is used, it should just truncate the sanity steps if needed, no?

Reply:
Yeah it's still has some issues with limit_val_batches and I think a better fix would be to setup num_sanity_val_steps as a list in Trainer itself rather than doing it here, and simple we can do a sum to get total sanity val steps.

Reply:
Does that means
-  When `num_sanity_val_steps == -1`:
  https://github.com/PyTorchLightning/pytorch-lightning/blob/0097630a95bddc48d6fb5d3b9a58aef2e8e89b22/tests/trainer/test_trainer.py#L802-L813
    -  `limit_val_batches != 0`: run `len(val_dataloader)` (could be `inf`) steps. (**independent**)
    -  `limit_val_batches == 0`, it shouldn't run any step. (**dependent**)
- When `num_sanity_val_steps >= 0`, the number of check steps should be affected by `limit_val_batches`.(**dependent**)

Reply:
I suggest in case of `num_sanity_val_steps == -1` it should be affected by `limit_val_batches` too.

Reply:
@rohitgr7 I like your suggestions. It is true, the trainer should compute these properties and the progress bars should only read them (and maybe sum them).

Reply:
Should I open another PR or keep this PR going? Should we use the same `num_sanity_val_steps` to save these values? (https://github.com/PyTorchLightning/pytorch-lightning/issues/2891#issuecomment-671054761)

Reply:
I am already working on it :)

Reply:
Thank you.